modules/core/src/lib/deck-picker.js

```diff
@@ -230,16 +224,14 @@ export default class DeckPicker {
     // Convert from canvas top-left to WebGL bottom-left coordinates
     // And compensate for pixelRatio
     const pixelRatio = this.pixelRatio;
-    const deviceLeft = Math.round(x * pixelRatio);
-    const deviceBottom = Math.round(this.gl.canvas.height - y * pixelRatio);
-    const deviceRight = Math.round((x + width) * pixelRatio);
-    const deviceTop = Math.round(this.gl.canvas.height - (y + height) * pixelRatio);
+    const [deviceLeft, deviceBottom] = cssToDevicePixels(this.gl, [x, y], true);
+    const [deviceRight, deviceTop] = cssToDevicePixels(this.gl, [x + width, y + height], true);
 
     const deviceRect = {
       x: deviceLeft,
       y: deviceTop,
-      width: deviceRight - deviceLeft,
-      height: deviceBottom - deviceTop
+      width: deviceRight - deviceLeft + 1,
+      height: deviceBottom - deviceTop + 1
```

Comment:
Why `+1` ? Make a comment?

Reply:
I made this change to fix a unit test, but looking deep into it, I can just adjust the test, the test allocates a 1 X 1, window and tests rect starting at (0, 0) and ending at (1, 1), which effectively needs 2 X 2 window.  The new utility methods I added in luma.gl , clamp down the final device pixels to actual bounds causing the test failure.
Updating the test to use 2 X 2 window, doesn't need this change and the test change below, but has another test failure in (Node only, browser is good, checking ...)

Reply:
Updated the PR to use newer luma utility method that returns range of valid pixels.
Old code was not correctly calculating `width` and `height`, we should add `1`, updated the comment, this change also needs update to picking test result.